sign.sh

```diff
@@ -62,24 +92,30 @@ signRelease()
       echo "$FILES" | while read -r f;
       do
         echo "Signing ${f}"
-        if ! "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t http://timestamp.globalsign.com/scripts/timestamp.dll "$f"; then
-          echo "RETRYWARNING: Failed to sign ${f} at $(date +%T): Possible timestamp server error - RC $? ... Retrying in 10 seconds"
-          sleep 10s
-          "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t http://timestamp.globalsign.com/scripts/timestamp.dll "$f"
-        fi        
+        for ((i = 0; i < ${#TIMESTAMPKEYS[@]}; i++))
+        do
+          # We still use a 5 second delay between attempts in case of a temporary localhost and/or local network condition
+          if ! "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t "${TIMESTAMPVALUES[i]}" "$f"; then
+            echo "RETRYWARNING: Failed to sign ${f} at $(date +%T): Possible timestamp server error at \"${TIMESTAMPVALUES[i]}\" - RC $? ... Trying new server in 5 seconds"            
+            sleep 5s
+          fi
+        done
       done
 
       # Sign .dll files
       FILES=$(find . -type f -name '*.dll')
-      echo "$FILES" | while read -r f;
-      do
-        echo "Signing ${f}"
-        if ! "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t http://timestamp.globalsign.com/scripts/timestamp.dll "$f"; then
-          echo "RETRYWARNING: Failed to sign ${f} at $(date +%T): Possible timestamp server error - RC $? ... Retrying in 10 seconds"
-          sleep 10s
-          "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t http://timestamp.globalsign.com/scripts/timestamp.dll "$f"
-        fi
-      done
+        echo "$FILES" | while read -r f;
+        do
+          echo "Signing ${f}"
+          for ((i = 0; i < ${#TIMESTAMPKEYS[@]}; i++))
+          do
+            # We still use a 5 second delay between attempts in case of a temporary localhost and/or local network condition
+            if ! "$signToolPath" sign /f "${SIGNING_CERTIFICATE}" /p "$SIGN_PASSWORD" /fd SHA256 /t "${TIMESTAMPVALUES[i]}" "$f"; then
+              echo "RETRYWARNING: Failed to sign ${f} at $(date +%T): Possible timestamp server error at \"${TIMESTAMPVALUES[i]}\" - RC $? ... Retrying in 5 seconds"
+              sleep 5s
+            fi
```

Comment:
Does need to break if it signs successfully?

Reply:
Almost certainly and it's why I made the comment about it looping signing multiple times https://github.com/AdoptOpenJDK/openjdk-build/pull/2405#issuecomment-766997573